c_src/sasl_auth.c

```diff
@@ -782,8 +801,9 @@ static ERL_NIF_TERM sasl_kinit(ErlNifEnv* env, int UNUSED(argc), const ERL_NIF_T
     }
     krb5_free_context(context);
 
-    enif_free(principal_mut);
-    enif_free(keytab_in);
+    enif_free(principal_char);
```

Comment:
Do you think we need a goto label here, as we have a few places where we might return quickly, but never free?

Reply:
fixed.